subprojects/distributions/src/integTest/groovy/org/gradle/DistributionIntegritySpec.groovy

```diff
@@ -191,6 +193,32 @@ class DistributionIntegritySpec extends DistributionIntegrationSpec {
         assert errors.empty
     }
 
+    @Requires(TestPrecondition.LINUX)
+    def "validate dependency archives"() {
+        given:
+        def unzipFolder = new File('/tmp/unzip_6.0-24')
+        def unzip = new File(unzipFolder, '/usr/bin/unzip')
+        if (!unzip.exists()) {
+            execute """
+                cd /tmp &&
+                wget 'http://ftp.debian.org/debian/pool/main/u/unzip/unzip_6.0-24_amd64.deb' &&
+                dpkg-deb -R unzip_6.0-24_amd64.deb ${unzipFolder.name}
+            """
```

Comment:
We shouldn't download things from the internet as part of an integration test. If we want to use this, we should have it installed on the CI executors via Salt. I think in our case it's okay to just run through each ZIP file using a `ZipFile`  and check if there are any entries with the same name.

Reply:
That's what I wanted to avoid, to make sure we don't reinvent our own zip bomb detection.
Would it be okay to download this from artifactory instead? Wanted to make the test self-contained, it's really difficult to know what's on which machine - and even more difficult to reproduce it locally, if we rely on CI state.

Reply:
I see two ways to tackle this:

1) Use `unzip 6.0+` and enforce it to be installed on the machines. You could have it present in some predefined directory, supplied maybe via an environmental variable similar to how we communicate the location of different JDKs with our testing infrastructure.

2) Write our own check using `ZipFile`.

Downloading things like this won't work for a number of reasons:

- Any external network resource used in a test adds the risk of flakiness that we should avoid whenever possible.
- The test should ideally run on Mac and Windows as well, neither of which support installing a Debian package.
- I'd feel uneasy about installing native binaries as part of a test.

I think 1) is overkill, so I'd go for 2), but I'm also fine with 1) if we care a lot about this from a security perspective. We should consult with @JLLeitschuh on this.

Reply:
Also, downloading over HTTP without verifying the artifact is a big red flag.

Reply:
I agree that this is a good check to have because we want to make sure our code works with unzip, but I question that this is the right way to do this.

Reply:
Done